### PR TITLE
return abci state's latest block number in rpc

### LINKED ISSUE #318: Query could fail with block number returned by `eth_blockNumber` api
__System info:__ ethermint main branch

__Steps to reproduce:__

1. start devnet
2. run script:
```python
import time

import web3

w3 = web3.Web3(web3.providers.HTTPProvider("http://localhost:1317"))
while True:
    no = w3.eth.block_number
    try:
        w3.eth.get_balance("0x57f96e6B86CdeFdB3d412547816a82E3E0EbF9D2", no)
    except:
        print("fail")
    else:
        print("success")
    time.sleep(1)
```

__Expected behavior:__ always success

__Actual behavior:__ fail sometimes

### PR DESCRIPTION
**Pull request by yihuang**
Closes #318

## Description

abci app state could lag behind tendermint reported latest block number.

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: ethereum/rpc/backend/backend.go [hunk 1]
```diff
@@ -69,15 +73,26 @@ func NewEVMBackend(logger log.Logger, clientCtx client.Context) *EVMBackend {
...
+	if l := len(blockHeight); l != 1 {
+		return 0, fmt.Errorf("unexpected '%s' header length; got %d, expected: %d", grpctypes.GRPCBlockHeightHeader, l, 1)
+	}
+
+	nBlockHeight, err := strconv.Atoi(blockHeight[0])
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse block height: %w", err)
+	}
+
+	return hexutil.Uint64(nBlockHeight), nil
```
Comment (fedekunze) [inline ethereum/rpc/backend/backend.go:95]:
```suggestion
	blockHeightHeader := header.Get(grpctypes.GRPCBlockHeightHeader)
	if headerLen := len(blockHeightHeader); headerLen != 1 {
		return 0, fmt.Errorf("unexpected '%s' gRPC header length; got %d, expected: %d", grpctypes.GRPCBlockHeightHeader, headerLen, 1)
	}

	height, err := strconv.FormatUint64(blockHeight[0], 10, 64)
	if err != nil {
		return 0, fmt.Errorf("failed to parse block height: %w", err)
	}

	return height, nil
```

File: ethereum/rpc/backend/backend.go [hunk 2]
```diff
@@ -69,15 +73,26 @@ func NewEVMBackend(logger log.Logger, clientCtx client.Context) *EVMBackend {
 	}
 }
 
-// BlockNumber returns the current block number.
+// BlockNumber returns the current block number in chain state.
```
Comment (fedekunze) [inline ethereum/rpc/backend/backend.go:76]: I'd add a note to outline the problem of using the height from blockchain info

**Review by fedekunze — APPROVED** [review]
ACK. 2 suggestions

File: ethereum/rpc/backend/backend.go (hunk 2 above)
Comment (yihuang) [reply ethereum/rpc/backend/backend.go]: done

**Review by fedekunze — APPROVED** [review]

**Pull request merged**